client/client_test.go

```diff
@@ -356,35 +355,36 @@ func TestKVClientEmptyValues(t *testing.T) {
 	}, &proto.PutResponse{})
 
 	getResp := &proto.GetResponse{}
-	kvClient.Call(proto.Get, proto.GetArgs(proto.Key("a")), getResp)
+	kvClient.Call(proto.GetArgs(proto.Key("a")), getResp)
 	if bytes := getResp.Value.Bytes; bytes == nil || len(bytes) != 0 {
 		t.Errorf("expected non-nil empty byte slice; got %q", bytes)
 	}
-	kvClient.Call(proto.Get, proto.GetArgs(proto.Key("b")), getResp)
+	kvClient.Call(proto.GetArgs(proto.Key("b")), getResp)
 	if intVal := getResp.Value.Integer; intVal == nil || *intVal != 0 {
 		t.Errorf("expected non-nil 0-valued integer; got %p, %d", getResp.Value.Integer, getResp.Value.GetInteger())
 	}
 }
 
-// TestKVClientPrepareAndFlush prepares a sequence of increment
-// calls and then flushes them and verifies the results.
-func TestKVClientPrepareAndFlush(t *testing.T) {
+// TestKVClientBatch runs a batch of increment calls and then verifies
+// the results.
+func TestKVClientBatch(t *testing.T) {
 	s := StartTestServer(t)
 	defer s.Stop()
 	kvClient := createTestClient(s.Addr)
 	kvClient.User = storage.UserRoot
 
+	calls := []*client.Call{}
 	replies := []*proto.IncrementResponse{}
 	keys := []proto.Key{}
 	for i := 0; i < 10; i++ {
 		key := proto.Key(fmt.Sprintf("key %02d", i))
 		keys = append(keys, key)
-		reply := &proto.IncrementResponse{}
-		replies = append(replies, reply)
-		kvClient.Prepare(proto.Increment, proto.IncrementArgs(key, int64(i)), reply)
+		call := client.IncrementCall(key, int64(i))
+		calls = append(calls, call)
+		replies = append(replies, call.Reply.(*proto.IncrementResponse))
 	}
 
-	if err := kvClient.Flush(); err != nil {
+	if err := kvClient.Run(calls...); err != nil {
```

Comment:
`Batch` implies running multiple items while it will efficiently run a single one. Seems odd, though I'm open to it.

Chatting with @bdarnell just now and I noted that most of our uses of `Prepare/Flush` are with transactions. Moving all of these uses over to building up an explicit slice of calls is onerous. What I'd like to do is change `KV.RunTransaction()` to create a `Txn` object which contains `Prepare/Flush`. Something like:

```
type Txn struct {
  kv *KV
  prepared []*Call
}

func (t *Txn) Prepare(call *Call) {
}

func (t *Txn) Flush() error {
}
```

This would also allow us to get rid of the oddity that you can call `RunTransaction` on the "transaction" `KV`.